File: integrations/runtime-bridges/python/bridge.py
"""Python runtime bridge template for AGenNext Kernels."""

class KernelBridge:
    def init(self, config: dict) -> None:
        self.config = config

    def invoke(self, action: dict) -> dict:
        return {"status": "todo", "action": action}

    def stream(self):
        yield {"status": "todo", "event": "stream-not-implemented"}

    def close(self) -> None:
        return None

File: integrations/runtime-bridges/node/bridge.ts
// Node.js / TypeScript runtime bridge template for AGenNext Kernels.
export class KernelBridge {
  private config: Record<string, unknown> = {};

  init(config: Record<string, unknown>): void {
    this.config = config;
  }

  invoke(action: Record<string, unknown>): Record<string, unknown> {
    return { status: "todo", action };
  }

  async *stream(): AsyncGenerator<Record<string, unknown>> {
    yield { status: "todo", event: "stream-not-implemented" };
  }

  close(): void {
    return;
  }
}
